main.nf

```diff
@@ -45,88 +82,696 @@ workflow.onComplete {
+		tuple(
+			row.group,
+			row.id,
+			row.sex,
+			row.type,
+		)
+	}
+
+	ch_expansionhunter_meta = ch_gatkcov_meta.filter { it ->
+		it[3] == "proband"
+	} // ch group, id, sex, type
+	ch_svvcf_to_bed_meta = ch_expansionhunter_meta
+
+	// BAM-start
+	// Check for .bam files in read1 and start from bam if any found.
+	ch_bam_start = ch_samplesheet
+		.filter {
+			row -> row.read1.endsWith("bam")
+		}
+		.map {
+			row ->
+			def group = row.group
+			def id = row.id
+			def bam = row.read1
+			def bai = row.read2
+			tuple(group, id, bam, bai)
+		}
+
+
+	copy_bam(ch_bam_start)
+	bamtoyaml(ch_bam_start)
+	ch_output_info = ch_output_info.mix(bamtoyaml.out.bamchoice_INFO)
+
+	ch_bam_start_dedup_dummy = Channel.empty()
+	if(params.run_melt) {
+		dedupdummy(ch_bam_start)
+		ch_bam_start_dedup_dummy = dedupdummy.out.dedup_dummy
+	}
+
+	ch_bam_bai = Channel.empty()
+	ch_bam_bai = ch_bam_bai.mix(copy_bam.out.bam_bai)
+
+	// PED //
+	ch_ped_input = ch_samplesheet
+		.filter { row -> row.type == "proband" }
+		.map { row ->
+			def group = row.group
+			def id = row.id
+			def type = row.type
+			def sex = row.sex
+			def father = row.father
+			def mother = row.mother
+			tuple(group, id, type, sex, mother, father)
+			}
+
+	create_ped(ch_ped_input)
+	ch_ped_base = create_ped.out.ped_base
+	ch_ped_fa = Channel.empty()
+	ch_ped_ma = Channel.empty()
+
+	ch_ped_trio = Channel.empty()  //TODO: better name for this channel
+	ch_ped_trio = ch_ped_trio.mix(ch_ped_base)
+	if(params.mode == "family" && params.assay == "wgs") {
+
+		ch_ped_fa = create_ped.out.ped_fa
+		ch_ped_ma = create_ped.out.ped_ma
+
+		ch_ped_trio = ch_ped_trio.mix(ch_ped_fa).mix(ch_ped_ma)
+		madeline(ch_ped_trio)
+
+		ch_versions = ch_versions.mix(madeline.out.versions.first())
+		ch_output_info = ch_output_info.mix(madeline.out.madde_INFO)
+
+	}
+
+	// FASTQ //
+	if (params.umi) {
+		fastp(ch_fastq)
+		ch_fastq = fastp.out.fastq_trimmed_reads
+		ch_versions = ch_versions.mix(fastp.out.versions.first())
+	}
+
+	// ALIGN //
+	//TODO: why do we have a params.align conditional anyway?
+	ch_dedup_stats = Channel.empty()
+	if (params.align) {
+		bwa_align(ch_fastq)
+		markdup(bwa_align.out.bam_bai)
+		ch_dedup_stats = ch_dedup_stats.mix(markdup.out.dedup_metrics)
+		ch_output_info = ch_output_info.mix(markdup.out.dedup_bam_INFO)
+		ch_bam_bai = ch_bam_bai.mix(markdup.out.dedup_bam_bai)
+
+		ch_versions = ch_versions.mix(bwa_align.out.versions.first())
+		ch_versions = ch_versions.mix(markdup.out.versions.first())
+	}
+
+	bqsr(ch_bam_bai)
+	ch_versions = ch_versions.mix(bqsr.out.versions.first())
+
+	// POST SEQ QC //
+	sentieon_qc(ch_bam_bai)
+	ch_versions = ch_versions.mix(sentieon_qc.out.versions.first())
+
+	ch_dedup_stats = ch_dedup_stats.mix(ch_bam_start_dedup_dummy)
+
+	sentieon_qc_postprocess(
+		sentieon_qc.out.sentieon_qc_metrics.join(ch_dedup_stats, by: [0,1])
+	)
+
+	ch_qc_json = ch_qc_json.mix(sentieon_qc_postprocess.out.qc_json)
+
+	// COVERAGE //
```

Comment:
We really need to divide things into subworkflows. 

I realize that will come later.